### DISCUSSION TITLE
Throwing custom error inside worker

### DISCUSSION THREAD

**Discussion by st3ffgv4** [poolifier/poolifier#2630] [Q&A] [answered]
Hi,
i want to know if this is a voluntary behaviour or not, or if it possibile to implement it or i didn't read well the docs.

The point is this: When inside a worker thread function a Custom error is thrown the execute catch Error type doesn't match, a default Error is catched.

index.js
```javascript
"use strict";
const {
  availableParallelism,
  FixedThreadPool,
  PoolEvents,
  DynamicThreadPool,
} = require("poolifier");
const CustomError = require("./CustomError");

const pool = new DynamicThreadPool(
  0,
  availableParallelism(),
  "./yourWorker.js",
  {
    errorHandler: (e) => console.error(e),
    onlineHandler: () => console.info("worker is online"),
  }
);

let resolved = 0;
const start = performance.now();
const iterations = 1;
for (let i = 1; i <= iterations; i++) {
  pool
    .execute()
    .then(() => {})
    .catch((err) => console.error(err instanceof CustomError));
}
``` 

yourWorker.js
```javascript
"use strict";
const { ThreadWorker } = require("poolifier");
const CustomError = require("./CustomError");

/**
 * @returns
 */
function yourFunction() {
  throw new CustomError("Test");
}

module.exports = new ThreadWorker(yourFunction);
``` 

CustomError.js
```javascript
module.exports = class CustomError extends Error {
  constructor(message, options) {
    super(message, options);

    this.name = "CustomError";
  }
};
``` 

The result is false.
Poolifier 4.3.0

**Comment by jerome-benoit** [comment]
Unfortunately, `Error` is not a structured-clonable object: test with `structuredClone()`. Only the Error destructuration `{ message, stack }` in strings can go into a message channel.

**Reply by jerome-benoit** [reply, marked as answer]
It's not an issue with structuredClone, but with cluster worker message passing. 
The code can properly handle the two cases and propagate worker side error unchanged when possible.  

Implemented in poolifier 4.4.1 for worker-threads pool only.

**Reply by st3ffgv4** [reply]
Thanks for investigating into this

**Comment by st3ffgv4** [comment]
```javascript
class HttpError extends Error {
  status;
  url;
  content;

  constructor(status, url, content) {
    super(`HTTP response code ${status} for ${url}`);

    this.status = status;
    this.url = url;
    this.content = content;
    this.name = "HttpError";
  }
}

module.exports = HttpError;
``` 

@jerome-benoit is not possible to convert this kind of error class back in the catch?
This kind of error is converted as a simple Error in the catch

Poolifier 4.4.1

**Reply by jerome-benoit** [reply]
The error is propagated as thrown in a `try {} catch {}` control flow construct. There's nothing more that can done on poolifier side. 

Try changing the error prototype or similar.
